### ISSUE TITLE
Encrypt/Decrypt

### ISSUE THREAD

**Issue by alexkunitsa** [gabriel/KBPGP#4]
I can't encrypt/decrypt messages. There is no error, but nothing happens. Method is called, but no callback.

```
 -[KBJSCore readFile:]:96: JSCore content:keybase.js, 1624538
 -[KBJSCore readFile:]:96: JSCore content:keybase-kbpgp-jscore.js, 19090
```

Can anybody help with that?

**Comment by alexkunitsa** [comment]
Looks like problem was with weak reference GHWeakSelf.

**Comment by gabriel** [comment]
yeah you have to keep a strong reference to KBPGP... For example, you'll
see the test cases have to do this.

it's a flaw... but to be honest the whole project, running pgp javascript
in a runtime with an objective-c, is a bit of a hack

this project is not ideal...

I'm working at Keybase now BTW and we're working on a objective-c to go
library which should make this library obsolete. In fact I've stopped using
it myself.

**Comment by alexkunitsa** [comment]
Looks like this is the best iOS PGP library. I I tried almost all possible cases) There is only one thing left that I need to implement - encrypt/decrypt files

**Comment by gabriel** [comment]
yeah theoretically it should be possible since kbpgp.js supports files:
https://keybase.io/kbpgp/docs/buffers

Though I probably won't have time to implement it myself anytime soon,
sorry :(
